appinventor/appengine/src/com/google/appinventor/shared/storage/StorageUtil.java

```diff
@@ -251,7 +252,11 @@ public static boolean isFontFile(String filePath)  {
    * Returns the URL for the given project file.
    */
   public static String getFileUrl(long projectId, String fileId) {
-    return GWT.getModuleBaseURL() + getFilePath(projectId, fileId);
+    // Note: Cannot call ode.GetModuleBaseURL() here because it is shared
+    // and Ode is not in scope on the server side.
```

Comment:
To this point, maybe it makes sense to put the logic in Ode in something like `ServerLayout` which is shared, and then it can be called both here and in Ode so we don't duplicate this logic (and have to keep track of the fact it is duplicated if we ever need to change it).